### PR TITLE
feat(relayer): Relay erc20 tokens

### PR DESCRIPTION
**Pull request by gshep**
Part of #75

### CODE REVIEW

File: relayer/src/erc20/mod.rs [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+    traits::Erc20Relay as _, Block, BlockBody, BlockHeader, BlockInclusionProof, Erc20Relay,
+    EthToVaraEvent, ExecutionPayload,
+};
+use futures::StreamExt;
+use gclient::{GearApi, WSAddress};
+use reqwest::Client;
+use sails_rs::{calls::*, events::*, gclient::calls::*, prelude::*};
+use std::cmp::Ordering;
+
+fn into_idl_execution_payload(
```
Comment (mertwole) [inline relayer/src/erc20/mod.rs:29]: Is it due to the fact that `sails` cannot encode it automatically?

**Review by mertwole — APPROVED** [review]

File: relayer/src/erc20/mod.rs (hunk 1 above)
Comment (gshep) [reply relayer/src/erc20/mod.rs:29]: Nope, it is consequence of regenerating types by sails. I requested the corresponding feature some time ago - https://github.com/gear-tech/sails/pull/538 . It is implemented however there is an issue - https://github.com/gear-tech/sails/issues/556 .

So for now we have to use these mapping funcs

File: relayer/Cargo.toml [hunk 2]
```diff
@@ -10,6 +10,11 @@ gear_proof_storage.workspace = true
 gear-rpc-client.workspace = true
 prover.workspace = true
 
+alloy.workspace = true
+alloy-consensus = { version = "0.2.1", default-features = false }
+alloy-eips = { version = "0.2.1", default-features = false }
+alloy-primitives = { version = "0.7.7", default-features = false }
```
[the remark this answers is not in the document]
Comment (gshep) [reply relayer/Cargo.toml:16]: ok

File: relayer/src/erc20/mod.rs (hunk 1 above)
Comment (mertwole) [reply relayer/src/erc20/mod.rs:29]: ok, then just open an issue in `gear-bridges` and link everything there

**Pull request merged**